great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -232,68 +236,103 @@ def _build_parameters(
+
+            round_decimals: int = self._get_round_decimals_using_heuristics(
                 metric_values=metric_values,
-                false_positive_rate=false_positive_rate,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
             )
 
-        min_value: Union[Number, float]
-        max_value: Union[Number, float]
+            metric_values = np.array(metric_values, dtype=np.float64)
+
+            if np.all(np.isclose(metric_values, metric_values[0])):
+                # Computation is unnecessary if distribution is degenerate.
+                lower_quantile = upper_quantile = metric_values[0]
+            elif sampling_method == "bootstrap":
+                lower_quantile, upper_quantile = self._get_bootstrap_estimate(
+                    metric_values=metric_values,
+                    false_positive_rate=false_positive_rate,
+                    domain=domain,
+                    variables=variables,
+                    parameters=parameters,
+                )
+            else:
+                lower_quantile, upper_quantile = compute_quantiles(
+                    metric_values=metric_values,
+                    false_positive_rate=false_positive_rate,
+                )
 
-        if round_decimals == 0:
-            min_value = round(float(lower_quantile))
-            max_value = round(float(upper_quantile))
-        else:
-            min_value = round(float(lower_quantile), round_decimals)
-            max_value = round(float(upper_quantile), round_decimals)
-
-        if lower_bound is not None:
-            min_value = max(min_value, lower_bound)
-        if upper_bound is not None:
-            max_value = min(max_value, upper_bound)
-
-        parameter_values: Dict[str, Any] = {
-            f"$parameter.{self.parameter_name}": {
-                "value": {
-                    "min_value": min_value,
-                    "max_value": max_value,
+            min_value: Union[Number, float]
+            max_value: Union[Number, float]
+
+            if round_decimals == 0:
+                min_value = round(float(lower_quantile))
+                max_value = round(float(upper_quantile))
+            else:
+                min_value = round(float(lower_quantile), round_decimals)
+                max_value = round(float(upper_quantile), round_decimals)
+
+            if lower_bound is not None:
+                min_value = max(min_value, lower_bound)
+            if upper_bound is not None:
+                max_value = min(max_value, upper_bound)
+
+            parameter_values: Dict[str, Any] = {
+                f"$parameter.{self.parameter_name}": {
+                    "value": {
+                        "min_value": min_value,
+                        "max_value": max_value,
+                    },
+                    "details": details,
                 },
-                "details": details,
-            },
-        }
+            }
```

Comment:
@jcampbell The above changes seem correct as long as they track the PR `https://github.com/great-expectations/great_expectations/pull/3692` precisely and the `Quentin` tests in that PR pass.  However, as mentioned above, this would be a short-term solution.  A long terms solution would include an architectural enhancement to enable interpreting and precessing the output of metrics of variable dimensions.

Reply:
roger that. I think this *is* from your work.